File: project.py
from project_module import project_object, image_object, link_object, challenge_object

p = project_object('youtube_thumbnailer', 'YouTube thumbnailer')
p.domain = 'http://www.aidansean.com/'
p.path = 'youtube_thumbnailer'
p.preview_image_ = image_object('http://placekitten.com.s3.amazonaws.com/homepage-samples/408/287.jpg', 408, 287)
p.github_repo_name = 'youtube_thumbnailer'
p.mathjax = True
p.links.append(link_object(p.domain, 'youtube_thumbnailer', 'Live page'))
p.introduction = 'This project takes a YouTube uri and provides the appropriate thumbnail images.'
p.overview = '''Occasionally it's useful to have the thumbnail for a YouTube video, and I often forget the exact path for doing this.  So I wrote this script that fetches the uri for the user.'''

p.challenges.append(challenge_object('The uri had to be parsed.', 'Since this is entirely Javascript based and in the browser there are no safety issues associated with the parsing.  Upon failure the page simply tells the user there was a problem.', 'Resolved'))

File: doc/project.php
<?php
include_once($_SERVER['FILE_PREFIX']."/project_list/project_object.php") ;
$github_uri   = "https://github.com/aidansean/youtube_thumbnailer" ;
$blogpost_uri = "http://aidansean.com/projects/?tag=youtube_thumbnailer" ;
$project = new project_object("youtube_thumbnailer", "YouTube thumbnailer", "https://github.com/aidansean/youtube_thumbnailer", "http://aidansean.com/projects/?tag=youtube_thumbnailer", "youtube_thumbnailer/images/project.jpg", "youtube_thumbnailer/images/project_bw.jpg", "This project takes a YouTube uri and provides the appropriate thumbnail images.", "Tools", "HTML,JavaScript,YouTube") ;
?>